models/nlp/classification/huggingface/inference.py

```diff
@@ -0,0 +1,56 @@
+import zipfile
+
+import torch
+import valohai
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from utils.serializers import get_serializer
+
+
+def load_model(model_path):
+    tokenizer = AutoTokenizer.from_pretrained(model_path)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = AutoModelForSequenceClassification.from_pretrained(model_path)
+    model.to(device)
+    return tokenizer, model, device
+
+
+def predict(text, tokenizer, model, device):
+    inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True)
+    inputs = {key: tensor.to(device) for key, tensor in inputs.items()}
+    outputs = model(**inputs)
+    logits = outputs.logits
+    predictions = torch.argmax(logits, dim=-1)
+    return predictions.item()
+
+
+def main():
+    model_path = valohai.inputs("model").path(process_archives=False)
+    if model_path.endswith(".zip"):
+        with zipfile.ZipFile(model_path, "r") as zip_ref:
+            zip_ref.extractall("/tmp/model")  # noqa: S108
+    else:
+        print(f"Model {model_path} is not a zip file. Aborting.")
+        return
+
+    tokenizer, model, device = load_model("/tmp/model")  # noqa: S108
+    data_path = valohai.inputs("data").path("*.txt")
+    rows = open(data_path).readlines()
+
+    results = []
+    for index, row in enumerate(rows):
+        text = row.strip()
+        prediction = predict(text, tokenizer, model, device)
+        results.append([text, prediction])
+
+        if (index + 1) % valohai.parameters("log_frequency").value == 0:
+            progress = (index + 1) / len(rows) * 100
+            print(f"Processed {index + 1}/{len(rows)} ({progress:.2f}%) items")
```

Comment:
You can let Python multiply your raw 0..1 progress by 100 and add a % with the `%` format specifier.
```suggestion
            progress = (index + 1) / len(rows)
            print(f"Processed {index + 1}/{len(rows)} ({progress:.2%}) items")
```